### ISSUE TITLE
rawkv: causality violation for atomic commands

### ISSUE THREAD

**Issue by pingyu** [tikv/tikv#13550]
### What version of TiKV are you using?

```bash
❯ tikv-server -V
TiKV
Release Version:   6.2.0-alpha
Edition:           Community
Git Commit Hash:   a68a44e09dd4ef7de59db22b11593e7abd4e94df
Git Commit Branch: heads/refs/tags/v6.3.0-alpha
UTC Build Time:    2022-09-27 10:22:06
Rust Version:      rustc 1.64.0-nightly (0f4bcadb4 2022-07-30)
Enable Features:   pprof-fp jemalloc mem-profiling portable sse test-engine-kv-rocksdb test-engine-raft-raft-engine cloud-aws cloud-gcp cloud-azure
Profile:           dist_release
```

### What operating system and CPU are you using?

CentOS

### Steps to reproduce

Run following test case:
```golang
func TestXX(..) {
	cli := rawkv.NewClient(...)
	cli.SetAtomicForCAS(true)

	Try(cli.Delete(ctx, key))

	// nil -> value0 -> value1
	{
		eg, ectx := errgroup.WithContext(ctx)
		eg.Go(WithGinkgoRecover(func() {
			Try(cli.Put(ectx, key, value0))
		}))
		eg.Go(WithGinkgoRecover(func() {
			var (
				err     error
				succeed bool
				retry   int
			)
			for !succeed && retry < CAS_MAX_RETRY {
				_, succeed, err = cli.CompareAndSwap(ectx, key, value0, value1)
				Expect(err).To(Succeed())
				retry += 1
			}
			Expect(succeed).To(BeTrue())
		}))
		Try(eg.Wait())
		actual := Try(cli.Get(ctx, key)).([]byte)
		Expect(actual).Should(Equal(value1), "key: %v, value0: %v, value1: %v", string(key), string(value0), string(value1))
	}
}
```

### What did you expect?

Test pass.

### What did happened?

Test failed
```
    key: b410000000000000240, value0: 541944245271235697, value1: 541944245271235698
    Expected
        <[]uint8 | len:18, cap:24>: 541944245271235697
    to equal
        <[]uint8 | len:18, cap:24>: 541944245271235698
```

**Comment by zeminzhou** [comment]
note: ts that is appended to key must be obtained after the value has been obtained.

**Comment by pingyu** [comment]
Closed by #13520
